Updating features proposal after latest discussions. (#48)
devcontainers/spec: proposals/devcontainer-features.md

Features can be defined by a `devcontainer-feature.json` file in the root folder of the feature. The file is optional for backwards compatibility but it is required for any new features being authored.

> By default features are executed by an order selected by the tool that implements the spec based in the following:

Reviewer: ```suggestion
By default features are executed by an order selected by the tool that implements the spec based on the following:
```

Author: Awesome thanks!

- `runsAfter` property defined as part of `devcontainer-feature.json`.
- id.

The tool uses the `runsAfter` propery to intelligently manage this order and ensure that if there are relationships betwen the features they are respected.

> In case an end user needs to control the order of execution we are adding the `overrideFeatureInstallOrder` to `devcontainer.json`

Author: 👍

> In case an end user needs to control the order of execution we are adding the `overrideFeatureInstallOrder` to `devcontainer.json`

Reviewer 2: ****

Features can be defined by a `devcontainer-feature.json` file in the root folder of the feature. The file is optional for backwards compatibility but it is required for any new features being authored.

By default, features are installed in an order selected by the implementing tool. 

> If any of the following properties in the feature's `devcontainer-feature.json, or the user's `devcontainer.json` are provided, the ordered indicated by the propert(ies) are respected.

Reviewer 3: ```suggestion
If any of the following properties are provided in the feature's `devcontainer-feature.json`, or the user's `devcontainer.json`, the order indicated by the propert(ies) are respected.
```

- `runsAfter` property defined as part of `devcontainer-feature.json`.

> - id.

Reviewer 3: ```suggestion
- `id`.
```

> 

Reviewer 3: Should this bulleted list include `overrideFeatureInstallOrder`? Does it need to mention `id`?

> The tool uses the `runsAfter` propery to intelligently manage this order and ensure that if there are relationships betwen the features they are respected.

Reviewer 3: ```suggestion
The tool uses the `runsAfter` property to intelligently manage this order and ensure that if there are relationships between the features, they are respected.
```

## devcontainer.json properties

> Features are referenced in `devcontainer.json` , where the `features` tag consists of an object tag starting with the id of the feature. and including the values of the options to pass to the feature itself.

Author: 👍

## devcontainer.json properties

Features are referenced in `devcontainer.json` , where the `features` tag consists of an object tag starting with the id of the feature and including the values of the options to pass to the feature itself.

The properties are:
| Property | Type | Description |
| :--- | :--- | :--- |

> | options | object | Type of the option. |

Reviewer 4: We don't need a separate property for options.

Author: Removed.

There are several things to keep in mind for an application that implements features:

> - The order of execution of features is determined by the application based on the `installAfter` property by feature authors and can be overriden by users if necesarry with the `overrideFeatureInstallOrder` in `devcontainer.json`.

Reviewer 3: ```suggestion
- The order of execution of features is determined by the application, based on the `installAfter` property used by feature authors. It can be overridden by users if necessary with the `overrideFeatureInstallOrder` in `devcontainer.json`.
```

> - Features are used to create an image that can be used to create a container or not.

Reviewer 3: What is the "or not" use case?

Author: I just meant that the code could be executed to just create an image for other purposes.

> - Parameters like `privileged`, `init` are included if just 1 feature requires them.

Reviewer 3: ```suggestion
- Parameters like `privileged` and `init` are included if just 1 feature requires them.
```

- Parameters like `capAdd`, `securityOp`  are concatenated.

> - ContainerEnv is added before the feature is executed as `ENV` commands in the docker file.

Reviewer 3: Should this be `containerEnv`?

```suggestion
- `containerEnv` is added before the feature is executed as `ENV` commands in the Dockerfile.
```
